### PR TITLE
Angular  7 Search Input Component and Clear

### PR DESCRIPTION
**Pull request by daanishnasir**
normal and pill search input, added clear button. 

## Description
Angular 7 Search Input

## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: angular/src/lib/search-input/search-input.component.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
+import { Component, OnInit, Input, Output, EventEmitter, Optional, Self} from '@angular/core';
+import { NG_VALUE_ACCESSOR, ControlValueAccessor, NgControl } from '@angular/forms';
```
Comment (mohammedzamakhan) [inline angular/src/lib/search-input/search-input.component.ts]: Remove unused imports

File: angular/src/lib/search-input/search-input.component.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+@Component({
+  selector: 'cui-search-input',
+  templateUrl: './search-input.component.html',
+  styles: []
+})
+
+export class SearchInputComponent implements ControlValueAccessor {
+
+  /** @prop Optional css class string | ''  */
+  @Input() public class: string = '';
```
Comment (mohammedzamakhan) [inline angular/src/lib/search-input/search-input.component.ts]: @pauljeter any idea about how to support ngClass for such components where the classes are applied inside the template

Comment (pauljeter) [reply angular/src/lib/search-input/search-input.component.ts]: No, for classes on internal components, we will expose a prop that we will pass down. i.e. the `inputClass` prop on line 23

File: angular/src/lib/input/input.component.html [hunk 3]
```diff
@@ -61,3 +64,87 @@
...
+        *ngIf="clear && value !== ''"
+        class="cui-button cui-button--36 cui-button--icon"
+        type="button"
+        aria-label="clear input"
+        (click)="handleClear()"
+      >
+        <span class="cui-button__children" style="opacity: 1;">
+          <i
+            class="cui-icon icon icon-clear-active_16 cui-input__icon-clear"
+            style="color: inherit;">
```
[the remark this answers is not in the document]
Comment (daanishnasir) [reply angular/src/lib/input/input.component.html]: removed

File: angular/src/lib/input/input.component.html [hunk 4]
```diff
@@ -61,3 +64,87 @@
   >
   </cui-input-helper>
 </div>
+
+
+<!-- With Clear Button -->
+<div *ngIf="clear" class="cui-input-group" [ngClass]="wrapperClasses">
```
[the remark this answers is not in the document]
Comment (daanishnasir) [reply angular/src/lib/input/input.component.html]: updated

**Pull request merged**